src/script/view_model/list/StartUIViewModel.js

```diff
@@ -377,6 +368,34 @@ class StartUIViewModel {
       this.searchResults.contacts(filteredResults);
 
       this.searchResults.groups(this.conversationRepository.getGroupsByName(normalizedQuery, isHandle));
+
+      if (!this.showOnlyConnectedUsers()) {
+        try {
+          const userEntities = await this.searchRepository.search_by_name(normalizedQuery, isHandle);
+
+          const isCurrentQuery = normalizedQuery === SearchRepository.normalizeQuery(this.searchInput());
+          if (isCurrentQuery) {
+            if (this.selfUser().inTeam()) {
```

Comment:
Huge block of business logic. Let's move it into a separate function so it can be easily tested.